clients/drcachesim/simulator/cache_simulator.h

```diff
@@ -60,12 +60,26 @@ class cache_simulator_t : public simulator_t {
     bool
     print_results() override;
 
+    int_least64_t
+    get_l1i_metric(metric_name_t metric, unsigned int core) const;
+    int_least64_t
+    get_l1d_metric(metric_name_t metric, unsigned int core) const;
+    int_least64_t
+    get_ll_metric(metric_name_t metric) const;
```

Comment:
In our usage we routinely have a 3-level cache with a per-core L2 in between L1{I,D} and LLC (created with a config file).  Possibilities:

+ Add a get_l2_metric().  To take in a core index, it would have to only support direct parents of L1* and so would not support arbitrary cache layouts.
+ Export an iterator over all caches (split into l1, llc, and "other"==l2 separate iterators?) which returns some kind of key (the cache name?) to then pass back to query a metric.
+ Take in the cache by name (the name from the config file) -- less usable by itself without an iterator.

Reply:
Thanks for the suggestions! Sorry for the delay, I had few days off. I like the flexibility of accessing stats in arbitrary cache model and also having really simple interface. I thought about having common function for all caches, something like `int_least64_t get_cache_metric(unsigned level, unsigned core, metric_t metric)`
Does separation for instruction and data caches makes sense only for L1 cache in any model ?

Reply:
I think the current code only supports i vs d at L1 but if it's easy not to assume that here that might help future support.

> something like `int_least64_t get_cache_metric(unsigned level, unsigned core, metric_t metric)`

Would this support asymmetric layouts?  I guess it would if you always walk from the core up `level` levels.

Reply:
I tried to make this code as flexible as possible. Current approach should work with split cache of the higher level in simple models, when given higher level split cache is an ancestor of L1 (for example L1i -> L2i).

Reply:
Can you provide example of some simple asymmetric layout ? My current approach is to walk up from L1 cache and I think it should work properly.

Reply:
> Can you provide example of some simple asymmetric layout ? My current approach is to walk up from L1 cache and I think it should work properly.

Never used or seen anything too weird with dr$sim, was just speculating about possible corner cases.  Such a walk should work with any layout though it seems.